### ISSUE TITLE
[BUG] Fluctuating Challenge Deadline

### ISSUE THREAD

**Issue by panjikharisma** [topcoder-platform/community-app#4076]
**Description**
The deadline of some challenges are changing (e.g. -/+ 10h) without the consent of PM/Copilot. It  has made [several members failing to submit/finish their submission properly](https://apps.topcoder.com/forums/?module=Thread&threadID=951482&start=0).

**To Reproduce - Sample Cases**
It's a server-side problem, and we don't know how to trigger this. But we've compiled a little list of sample cases on when these things occur:

- [Infant Nutrition UI Design Challenge - Checkpoint Deadline - Feb 2020](https://www.topcoder.com/challenges/30116805)
   Noticed by: kharm, rajeshrathod, eriantoongko, joni7sunny, iqbalhood, lemonnine
   > User story (kharm): Challenge started, challenge stalled, deadline changed around +10h and challenge resumed (not sure changed or resumed first), ...passed for days..., deadline changed -10h (or back to normal around 1 day before the checkpoint deadline).
- [Novus Driving Mobile Application Design Challenge - Checkpoint Deadline - Feb 2020](https://www.topcoder.com/challenges/30114365)
   Noticed by: kharm
   > Notes (kharm): Not sure about this one, it's either I'm the only one who noticed it and other competitors just brushed it off (as no one raised this issue right off the bat), or I'm the one who remember it wrong. In short I fell asleep and the checkpoint deadline was already over (-/+10h from my mental memory).
- [Disrupt3rs Design Direction Challenge 39 Hour - Final Deadline - May 2019](https://www.topcoder.com/challenges/30090883)
   Noticed by: S-M, stack1819995, iamtong, rajeshrathod
   > User story (DaraK-copilot): Posted checkpoint feedback, timeline was alright, went to sleep, woke up in the morning the timeline was fine as well but members said during the night it had changed with 4-5 hours less and then back to normal.

**Expected behavior**
Challenge deadline shouldn't be changing without PM/Copilot consent.

**Screenshots**
No screenshots, but some members direct report on the latest event could be seen [here - Infant Nutrition Welcome thread](https://apps.topcoder.com/forums/?module=Thread&threadID=951482&start=0).

**Desktop (mine):**
 - OS: MacOS
 - Browser: Chrome
 - Version 80.0.3987.122 

**Additional context**
- This bug was known for at least for nearly 1 year in the community ([Disrupt3rs challenge](https://www.topcoder.com/challenges/30090883) was May 2019)
- Several members testify they've faced it other times before - chekspir [post](https://apps.topcoder.com/forums/?module=Message&messageID=2393802), DaraK [post](https://apps.topcoder.com/forums/?module=Message&messageID=2394601)
- Thread focusing on this issue & asking members to share their related experience with this bug could be found [here - General Discussions,  Fluctuating Challenge Deadline](https://apps.topcoder.com/forums/?module=Thread&threadID=951914&start=0).

Hope you can help us on this, Thank you!

\- kharm

**Comment by panjikharisma** [comment]
Another sample case with API response + before & after print screen:
* [Woodburn Insurance CMS Mobile Site Challenge - Checkpoint](https://www.topcoder.com/challenges/30119207)
   Noticed by: kharm, petetebe, idblack
   > During checkpoint review, the deadline fluctuated from into a shorter period of time.
   >
   > Screenshots:
   > [Initial deadline - March 13 - 15:29:38 GMT+7 - Just few hours after the contest started](https://drive.google.com/open?id=1eqJWQ-Y8aWNdncgcdhqeK7FVgsfu_Bcg)
   > [Bug appeared - March 14 - 14:40:51 GMT+7](https://drive.google.com/open?id=1WJXp3YpXvXLKUsx-xw39kp17w6aRCRkZ)
   > _* Notice that the order of the deadline fields were also changed_
   >
   > Forum link where the corresponding event discussed: https://apps.topcoder.com/forums/?module=Thread&threadID=952482&start=0&mc=25#2396848
   > 
   > API response during March 14, around the screenshot taken (from https://api.topcoder.com/v4/challenges/?filter=id%3D30119207):
   ```
   {"id":"-2b5f5b6b:170d75c1c8f:-50d7","result":{"success":true,"status":200,"metadata":{"totalCount":1,"allChallengesCount":1,"myChallengesCount":1,"openChallengesCount":1,"ongoingChallengesCount":0},"content":[{"updatedAt":"2020-03-13T17:30Z","createdAt":"2020-03-13T04:37Z","createdBy":"22738837","updatedBy":"22738837","status":"ACTIVE","track":"DESIGN","subTrack":"WIDGET_OR_MOBILE_SCREEN_DESIGN","name":"Woodburn Insurance CMS Mobile Site Challenge","reviewType":"INTERNAL","id":30119207,"forumId":732659,"numSubmissions":1,"numRegistrants":44,"numSubmitters":1,"registrationStartDate":"2020-03-13T05:45:00.485Z","registrationEndDate":"2020-03-17T13:59:00.000Z","checkpointSubmissionEndDate":"2020-03-14T14:58:00.000Z","submissionEndDate":"2020-03-21T22:15:00.000Z","totalCheckpointPrize":500.0,"totalPrize":2750.0,"isPrivate":false,"upcomingPhase":{"id":1283841,"phaseType":"Checkpoint Screening","phaseStatus":"Scheduled","scheduledStartTime":"2020-03-14T14:58:00.000Z","scheduledEndTime":"2020-03-14T23:59:00.000Z","duration":32460000},"projectId":25802,"currentPhases":[{"id":1283838,"phaseType":"Registration","phaseStatus":"Open","scheduledStartTime":"2020-03-13T05:45:00.485Z","scheduledEndTime":"2020-03-17T13:59:00.000Z","actualStartTime":"2020-03-13T05:45:00.485Z","fixedStartTime":"2020-03-13T05:00:00.000Z","duration":375240000},{"id":1283839,"phaseType":"Checkpoint Submission","phaseStatus":"Open","scheduledStartTime":"2020-03-13T06:23:54.753Z","scheduledEndTime":"2020-03-14T14:58:00.000Z","actualStartTime":"2020-03-13T06:23:54.753Z","duration":117305247},{"id":1283840,"phaseType":"Submission","phaseStatus":"Open","scheduledStartTime":"2020-03-13T17:30:13.736Z","scheduledEndTime":"2020-03-21T22:15:00.000Z","actualStartTime":"2020-03-13T17:30:13.736Z","duration":708300000}],"submissionViewable":false,"allPhases":[{"id":1283836,"phaseType":"Specification Submission","phaseStatus":"Closed","scheduledStartTime":"2020-03-13T04:59:32.232Z","scheduledEndTime":"2020-03-13T05:01:28.161Z","actualStartTime":"2020-03-13T04:59:32.232Z","actualEndTime":"2020-03-13T05:01:28.161Z","fixedStartTime":"2020-03-11T03:00:00.000Z","duration":3600000},{"id":1283837,"phaseType":"Specification Review","phaseStatus":"Closed","scheduledStartTime":"2020-03-13T05:02:01.046Z","scheduledEndTime":"2020-03-13T05:39:26.594Z","actualStartTime":"2020-03-13T05:02:01.046Z","actualEndTime":"2020-03-13T05:39:26.594Z","duration":1800000},{"id":1283838,"phaseType":"Registration","phaseStatus":"Open","scheduledStartTime":"2020-03-13T05:45:00.485Z","scheduledEndTime":"2020-03-17T13:59:00.000Z","actualStartTime":"2020-03-13T05:45:00.485Z","fixedStartTime":"2020-03-13T05:00:00.000Z","duration":375240000},{"id":1283839,"phaseType":"Checkpoint Submission","phaseStatus":"Open","scheduledStartTime":"2020-03-13T06:23:54.753Z","scheduledEndTime":"2020-03-14T14:58:00.000Z","actualStartTime":"2020-03-13T06:23:54.753Z","duration":117305247},{"id":1283840,"phaseType":"Submission","phaseStatus":"Open","scheduledStartTime":"2020-03-13T17:30:13.736Z","scheduledEndTime":"2020-03-21T22:15:00.000Z","actualStartTime":"2020-03-13T17:30:13.736Z","duration":708300000},{"id":1283841,"phaseType":"Checkpoint Screening","phaseStatus":"Scheduled","scheduledStartTime":"2020-03-14T14:58:00.000Z","scheduledEndTime":"2020-03-14T23:59:00.000Z","duration":32460000},{"id":1283842,"phaseType":"Checkpoint Review","phaseStatus":"Scheduled","scheduledStartTime":"2020-03-14T23:59:00.000Z","scheduledEndTime":"2020-03-15T11:17:00.000Z","duration":40680000},{"id":1283843,"phaseType":"Screening","phaseStatus":"Scheduled","scheduledStartTime":"2020-03-21T22:15:00.000Z","scheduledEndTime":"2020-03-22T02:15:00.000Z","duration":14400000},{"id":1283844,"phaseType":"Review","phaseStatus":"Scheduled","scheduledStartTime":"2020-03-22T02:15:00.000Z","scheduledEndTime":"2020-03-28T02:15:00.000Z","duration":518400000},{"id":1283845,"phaseType":"Approval","phaseStatus":"Scheduled","scheduledStartTime":"2020-03-28T02:15:00.000Z","scheduledEndTime":"2020-04-02T02:15:00.000Z","duration":432000000}],"prizes":[1700.0,800.0,250.0],"events":[{"eventId":3452,"eventName":"2020 Topcoder(R) Open","projectId":25802}],"isTask":false,"isRegistered":false}]},"version":"v3"}
   ```

@Oanh-and-only-Oanh : Just to let you know, I've added a new sample case with a better documentation. Let me know if you need some critical data to help, so I can record that also in case I found another case and still have some willpower left to solve this, Lol.

**Comment by SathyaJayabal** [comment]
@Oanh-and-only-Oanh , looks like the v4 challenge api response had the incorrect data, based on api response data posted above, This could have been fixed in V5 challenge api. Let us wait to see if this still happens in V5 challenge api.

@rootelement , thoughts?

**Comment by Oanh-and-only-Oanh** [comment]
@panjikharisma, please let us know if you're still experience this issue. Thank you.

**Comment by panjikharisma** [comment]
Hi @Oanh-and-only-Oanh & @SathyaJayabal : I'll be paying attention to the schedules for several challenges in the next 7 days, and will get back to you next week. This bug is something that come out of nowhere in the previous records. Thanks!

**Comment by panjikharisma** [comment]
@Oanh-and-only-Oanh & @SathyaJayabal just a little update, I'm on the track on this one currently (4 days - around 10/11 challenges tracked). Will get back to you after after I got 7 day screenshots.

**Comment by panjikharisma** [comment]
Hi @Oanh-and-only-Oanh & @SathyaJayabal , here are some updates on this issue.

I've tracked 11 challenges through out Sep 11-Sep 17 2020 GMT+7, once a day at a random time. Here are the screenshots for each challenge, both on the Challenge List & Challenge Detail:
https://drive.google.com/drive/folders/11f49Av9D2l1euqNe-Euh6b046TXDj7Tm?usp=sharing

Overall I didn't see any significant fluctuations. Most are good, some have weird fluctuations but only in minutes, for complete information you may see the list below:

**Airtel** [Challenge Link](https://www.topcoder.com/challenges/94ab20a7-5354-403f-90d1-a10bbcf6ed93)
> Things looking good, except ‘1 minute decrease’ (albeit some with 1 day increase also) on deadline each day while checkpoint review and early final submission round. Seems because late review process (or extension side effect).

**Design Practice** [Challenge Link](https://www.topcoder.com/challenges/79db41f6-fb50-4bd7-a9bc-d7881c69388e)
> Things are good, not a single change on the deadline.

**Dragonet Data Management** [Challenge Link](https://www.topcoder.com/challenges/b635316a-a41d-490e-915f-5dd21a1e024f)
> Overall seems good. There are some changes, but it seems there was copilot interference as it is already in the review phase and the changes is nearby the end of the review phase.

**Dragonet KRI** [Challenge Link](https://www.topcoder.com/challenges/b0a6fd11-638b-4087-b805-99b17558af17)
> Again some changes, but most likely copilot interference.

**Odyssey Account**  [Challenge Link](https://www.topcoder.com/challenges/b5361fac-3e75-4d5e-9c41-644e25c1dbe7)
> Some changes during checkpoint review looks deliberate.

**Odyssey Alert** [Challenge Link](https://www.topcoder.com/challenges/e0bb3961-3e09-4051-8ee2-c3b4c7bcb117)
> Looks good

**Opportunity Management** [Challenge Link](https://www.topcoder.com/challenges/07823419-4417-441a-aaae-9dce5564606a)
> Looks good

**Slingshot** [Challenge Link](https://www.topcoder.com/challenges/c84d6481-656d-435c-9bcf-fa851ac5f45d)
> 8 minute addition for final round deadline during the last day. Probably intentional from the copilot for preventive measure.

**Topcoder TaaS** [Challenge Link](https://www.topcoder.com/challenges/9820cdb3-c316-4649-a3e3-4a98d0abdc7a)
> A slight changes (minutes) for registration, checkpoint submission etc in several days. However I’m not sure how to interpret this one.

**TELUS** [Challenge Link](https://www.topcoder.com/challenges/25497665-9303-4d3d-bb8c-bd2a6a08a9be)
> There are 1-3 minutes fluctuations per each ‘change/phase’.

**Topcoder Mask** [Challenge Link](https://www.topcoder.com/challenges/119fa2c9-e8d1-48a4-98e5-8d48fe639468)
> 4/5 minute addition during the final round, seems deliberate.

Here are some cases where I found a little bit weird (just a little), in case you are interested:
[Airtel Screenshots](https://drive.google.com/drive/folders/1pm0QlUPCh_dBwsNO_GJS6AAbMfe9Qjrp?usp=sharing) (1 minute changes, seems server input error), [Topcoder TaaS Screenshots](https://drive.google.com/drive/folders/1v054P_r48oQxv4kyZSxq6QtgyvCFjtE0?usp=sharing) and [TELUS Screenshots](https://drive.google.com/drive/folders/10I64wt-DRjTMxBotem5RolJ-6GZ8B-Kc?usp=sharing)



**The bottom line** I think you may close this issue for now, and I'll write some notes on the thread on Topcoder forum also, just in case one day this issue appear again, so they may directly report to this Github thread instead.


Thank you.

**Comment by panjikharisma** [comment]
P.s.: You may use/save the screenshots also for future use, I just do an analysis at a glance just now :)
